db/src/db.rs

```diff
@@ -189,8 +197,30 @@ fn get_user_version(conn: &rusqlite::Connection) -> Result<i32> {
         .chain_err(|| "Could not get_user_version")
 }
 
+/// Returns the rusqlite version number for a version string
+///
+/// Rusqlite version numbers are created with a 0 for each '.' in the version string
+/// and with each part of the version number, except for the major version, at least 2 characters long.
+/// If a part of the version string is only one character long then it is padded to 2 characters with 0's.
+fn version_number_from_string(version_string: &'static str) -> i32 {
+    let split = version_string.split(".");
+    let mut version_number: String = "".to_owned();
+    for s in split {
+        version_number.push_str("0");
+        if s.chars().count() < 2 {
+            version_number.push_str("0")
+        }
+        version_number.push_str(s);
+    }
+    return version_number.parse::<i32>().unwrap();
+}
+
 // TODO: rename "SQL" functions to align with "datoms" functions.
 pub fn create_current_version(conn: &mut rusqlite::Connection) -> Result<DB> {
+    if rusqlite::version_number() < version_number_from_string(MIN_SQLITE_VERSION) {
```

Comment:
Why not just define `MIN_SQLITE_VERSION` as `3008000` and do that here, rather than adding runtime parsing code?

I think it's quite common to express SQLite versions in this way, so I'm very happy to simply hardcode the simple thing:

```
const MIN_SQLITE_VERSION = 3008000;    // 3.8.0
```